ethcore/evm/src/interpreter/mod.rs

```diff
@@ -412,6 +422,21 @@ impl<Cost: CostType> Interpreter<Cost> {
 				};
 				self.reader.position = pos;
 			},
+			InstructionResult::JumpToSubroutine(position) => {
+				if self.valid_subroutine_destinations.is_none() {
```

Comment:
When we fetch `valid_subrountine_destinations`, we can set `valid_jump_destinations` at the same time, and vice versa, because those two values are always computed at the same time. Should slightly improve performance.